extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -942,14 +957,31 @@ async function handleSelectedAccountChanged(
 			}
 		}
 		subscriptionDropdown.values = response.subscriptions.map(subscription => {
-			const displayName = `${subscription.name} (${subscription.id})`;
+			const displayName = getSubscriptionDisplayString(subscription);
 			subscriptionValueToSubscriptionMap.set(displayName, subscription);
 			return displayName;
 		}).sort((a: string, b: string) => a.toLocaleLowerCase().localeCompare(b.toLocaleLowerCase()));
 		const selectedSubscription = subscriptionDropdown.values.length > 0 ? subscriptionValueToSubscriptionMap.get(subscriptionDropdown.values[0]) : undefined;
 		await handleSelectedSubscriptionChanged(context, selectedAccount, selectedSubscription, resourceGroupDropdown);
 	} catch (error) {
-		vscode.window.showErrorMessage(localize('azure.accounts.unexpectedSubscriptionsError', "Unexpected error fetching subscriptions for account {0} ({1}): {2}", selectedAccount?.displayInfo.displayName, selectedAccount?.key.accountId, getErrorMessage(error)));
+		await showAzureAccessError(selectedAccount, error, localize('azure.accounts.unexpectedSubscriptionsError', "Unexpected error fetching subscriptions for account {0}: {1}", getAccountDisplayString(selectedAccount), getErrorMessage(error)));
+	}
+}
+
+function getSubscriptionDisplayString(subscription: azureResource.AzureResourceSubscription) {
+	return `${subscription.name} (${subscription.id})`;
+}
+
+async function showAzureAccessError(selectedAccount: azdata.Account, error: any, unexpectedErrorMessage: string) {
+	switch (await getAccountStatus(selectedAccount)) {
+		case AccountStatus.notFound:
+			vscode.window.showErrorMessage(localize('azure.accounts.accountNotFoundError', "The selected account '{0}' is no longer available. Kindly add it back by signing in.\n Error Details: {1}.", getAccountDisplayString(selectedAccount), getErrorMessage(error)));
+			break;
+		case AccountStatus.isStale:
+			vscode.window.showErrorMessage(localize('azure.accounts.accountStaleError', "The access token for selected account '{0}' is no longer valid. Kindly sign in again to refresh your credentials.\n Error Details: {1}.", getAccountDisplayString(selectedAccount), getErrorMessage(error)));
```

Comment:
`Please click the sign in button and refresh the account or select a different account`

Reply:
done